apiserver/src/api/mod.rs

```diff
@@ -229,6 +239,30 @@ pub async fn run_server<T: 'static + BottlerocketShadowClient>(
     Ok(())
 }
 
+// Certificate is refreshed periodically (default 60 days). Once certificate is renewed, apiserver
+// needs to stop the server for reloading new certificate.
+// We cache the certificate initially when brupop starts server, and compare to update-to-date certificate periodically.
+// If they aren't match, we recognize it as new certificate is coming out, so the server needs to be restarted.
+async fn reload_certificate(
+    server: Server,
+    public_key_path: &str,
+    certificate_cache: Vec<u8>,
+) -> Result<()> {
+    loop {
+        let current_certificate =
+            read_certificate(public_key_path).context(error::ReadCertificateFailedSnafu)?;
+        if current_certificate != certificate_cache {
+            event!(
+                Level::INFO,
+                "Certificate has been renewed, restarting server to reload new certificate"
+            );
+            server.stop(true).await;
+            std::process::exit(0)
```

Comment:
I don't like to see a process exit buried so deeply in a program. Ideally you would return a bool (or enum) that would say whether or not an exit is needed. Bubble that all the way up to main if possible. Ideally only main should have process exits IMO.

Reply:
After testing, I think we don't need `process exit` since `server.stop(true).await;` will restart apiserver. I'll just remove this `std::process::exit(0)`. thanks